PowerShell Contributing.md

```diff
@@ -0,0 +1,479 @@
+
+### Updating your fork
+
+So you have forked one of the GitHub repos, cloned it, tried stuff, but never created a PR. And now that fork is six months behind its ancester. How to update it? I'll describe how to do this for the PowerShell repo, this works the same for the other repos.
+
+To update your GitHub fork, you will first update your local clone, and for that, you need to tell your local repository to which remote to look. 
+
+- See which remotes are configured for your local repo: `git remote -v`
+- If there is an entry with `https://github.com/PowerShell/PowerShell.git` in it, remember its name. It will usually be called 'upstream'
+- If there is no entry present with that url, add it by typing:
+`git remote add upstream https://github.com/PowerShell/PowerShell.git`
+- Make sure there is an entry 'origin' with the url of your own GitHub url: `https://github.com/PowerShell/PowerShell.git` 
+
+Make sure you have no uncommitted work on your branch. Stash or revert any uncommitted changes, and checkout to the master branch:
+
+- To stash changes, go `git stash`
+- To checkout master, do `git checkout master`
+
+Now we need to check for changes and pull them in.
+
+- Tell git to find out if there are changes in any of its remotes: `git fetch --all -p`
+- Tell git to rebase your local repo on the 'upstream' remote (replace the name 'upstream' if necessary):
+`git rebase upstream/master`
+
+Rebasing is the way to tell git to replay your work on top of the other changes that were merged to master in the mean time. So now we have pulled in all changes from the original repo into our clone locally. These changes need to be pushed to your GitHub fork.
+
+- Tell git to push local changes to origin: `git push` or `git push origin/master`.
+
+For more detailed instructions with screenshots, see [here](https://info.sapien.com/index.php/version-control/github-how-to-update-your-fork).
+
+### Submitting your contribution 
+
+Do small git commits while you work, and don't forget to git push the work on your branch every once in a while.
+When you are satisfied with your work, you can create a Pull Request.
+You can create a PR from your own GitHub on the branch you created for you work on this.
+It helps to reference your original Issue where you asked for guidance what to work on, use a '#' sign and put the number of your Issue behind it.
+Fill any fields the PR requests from you.
+
+Once you have created a PR, the automated CI process will start to run.
+It will run ALL tests, on Windows, Linux and OSX.
+You can follow the status of the running tests in your PR.
+It takes around 20 minutes or so to finish.
+Every time you push a new commit to your branch that you created the PR from, it will trigger a new CI run.
+
+If you want to signal the maintainers that you are working on stuff, but are not ready yet, you can rename your PR to start with `WIP:` for Work In Progress.
```

Comment:
Github also has a native WIP option for PRs when you first open them too (it's a drop-down option on the Submit button for PRs)

Reply:
I could not for the life of me find it _after_ having already created the PR -and I was fairly sure it existed before. This explains things, lol